### PR TITLE
Feature/dlit 1053 all certificates

### PR DESCRIPTION
**Pull request by hafthorarinsson**

### CODE REVIEW

File: src/Pages/Certificates/EditProperties.tsx [hunk 1]
```diff
@@ -1,75 +1,143 @@
...
+    if (dictionaryType === 'rating') {
+      CertificatesApi.GetAllCertificationRatings().then(res => {
+        setRatingTypes(res.data)
+      })
     }
   }
 
+  const addProperty = (newProperty: string) => {
+    var propertyToAdd
+    if (tab_value === 0) {
```
Comment (helgigretar) [inline src/Pages/Certificates/EditProperties.tsx:52]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/EditProperties.tsx [hunk 2]
```diff
@@ -1,75 +1,143 @@
...
+      propertyToAdd = { aircraft_category: newProperty, created_by_employee_id: currentUser.user_id }
+      CertificatesApi.CreateNewAircraftCategory(propertyToAdd).then(res => {
+        refresh('aircraft')
+        setNewPropertyInput('')
+      })
+    }
+    if (tab_value === 1) {
```
Comment (helgigretar) [inline src/Pages/Certificates/EditProperties.tsx:59]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/EditProperties.tsx [hunk 3]
```diff
@@ -1,75 +1,143 @@
...
+      propertyToAdd = { rating: newProperty, created_by_employee_id: currentUser.user_id }
+      CertificatesApi.CreateNewCertificateRating(propertyToAdd).then(res => {
+        refresh('rating')
+        setNewPropertyInput('')
+      })
+    }
+  }
+  const deleteProperty = (id: number) => {
+    if (tab_value === 0) {
```
Comment (helgigretar) [inline src/Pages/Certificates/EditProperties.tsx:68]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/EditProperties.tsx [hunk 4]
```diff
@@ -1,75 +1,143 @@
...
+    }
+    if (tab_value === 1) {
+      CertificatesApi.RemoveCertificateRatings(id).then(res => {
+        refresh('rating')
+      })
+    }
+  }
 
+  const onChangeTabValue = (tab_value: number) => {
+    if (tab_value === 0) {
```
Comment (helgigretar) [inline src/Pages/Certificates/EditProperties.tsx:81]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/EditProperties.tsx [hunk 5]
```diff
@@ -1,75 +1,143 @@
...
+      refresh('aircraft')
+    }
+    if (tab_value === 1) {
```
Comment (helgigretar) [inline src/Pages/Certificates/EditProperties.tsx:84]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/History.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                        <Typography margin={[1, 0, 0, 2]} align="left" text={row.mechanic_full_name}></Typography>
+                    </Grid>
+                </TableCell>
+                <TableCell>{row.created_date}</TableCell>
+                <TableCell>{row.invalid_date} </TableCell>
+            </TableRow>
+        )
+    }
+    const OnOrderByTitles = (given_order_by_text: string) => {
+        let temp_ascending = true;
```
Comment (helgigretar) [inline src/Pages/Certificates/History.tsx:68]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/History.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        if (order_by_text === given_order_by_text) { temp_ascending = ascending === false ? true : false; }
```
Comment (helgigretar) [inline src/Pages/Certificates/History.tsx:69]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/History.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                        <Grid xs={2}>
+                            <Button variant="contained" autoFocus color="primary" size="large" onClick={() => searchCertificates()}>Search</Button>
+                        </Grid>
+                    </Grid>
+                    <Grid>
+                        <Table
+                            headers={["Employee Name", "Created", "Expired"]}
+                            rows={certificatesResult}
+                            generateBody={(row: any, index: number) => generateTableBody(row, index)}
+                            default_rows_per_page={100}
```
Comment (helgigretar) [inline src/Pages/Certificates/History.tsx:110]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/WaitingForApproval.tsx [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    const [selectedCertificate, setSelectedCertificate] = useState<CertificateData>();
+    const [order_by_text, setOrderByText] = useState("created_date");
+    const [ascending, setAscending] = useState(true);
+    const [approver, setApprover] = useState<boolean>(true)
+    const store = useSelector((state: RootState) => state);
+    const userGroups = store.user.me.user_clearence_groups;
+    const classes = useStyles();
+
+    useEffect(() => {
+        if(userGroups !== undefined) {
```
Comment (helgigretar) [inline src/Pages/Certificates/WaitingForApproval.tsx:37]: Missing indent.

File: src/Pages/Certificates/WaitingForApproval.tsx [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+                setSelectedCertificate(res.data)
+            })
+        }
+    }
+    const closeCertificate = () => {
+        refresh()
+        setSelectedCertificate(undefined);
+    }
+    const OnOrderByTitles = (given_order_by_text: string) => {
+        let temp_ascending = true;
```
Comment (helgigretar) [inline src/Pages/Certificates/WaitingForApproval.tsx:83]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/WaitingForApproval.tsx [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    return (
+        <Grid>
+            {selectedCertificate === undefined ? <Grid>
+                <Grid>
+                    {certificatesPending !== undefined ?
+                        <Table
+                            headers={["Employee Name", "Approval requested date", ""]}
+                            rows={certificatesPending}
+                            generateBody={(row: any, index: number) => generateTableBody(row, index)}
+                            default_rows_per_page={100}
```
Comment (helgigretar) [inline src/Pages/Certificates/WaitingForApproval.tsx:98]: we are not using snake case for variables in typescript. Use Camel case

File: src/Pages/Certificates/index.tsx [hunk 12]
```diff
@@ -1,23 +1,42 @@
...
 
 function Certificates() {
+  const store = useSelector((state: RootState) => state);
+  const [userType, setUserType] = useState('normal')
+  const userGroups = store.user.me.user_clearence_groups;
   const [tab_value, setTabValue] = useState(0);
+
+  useEffect(() => {
+    if(userGroups !== undefined) {
+    if (userGroups?.indexOf("CertificateEditor") > -1 || userGroups.indexOf("CertificateApprover") > -1) {
```
Comment (helgigretar) [inline src/Pages/Certificates/index.tsx:19]: Missing indent

File: src/Pages/Certificates/index.tsx [hunk 13]
```diff
@@ -1,23 +1,42 @@
...
+  }, [userGroups])
+
+
   return (
     <Grid xs={12} item>
       <Tabs
-        tab_names={["All Certificates", "Waiting for Approval", "History", "Edit properties"]}
+        tab_names={userType !== 'normal' ? ["All Certificates", "Waiting for Approval", "History", "Edit properties"] : ["All Certificates"]}
         onChangeTabValue={(tab_value: number) => setTabValue(tab_value)}
         tab_value={tab_value}
```
Comment (helgigretar) [inline src/Pages/Certificates/index.tsx:34]: we are not using snake case for variables in typescript. Use Camel case

**Review by helgigretar — COMMENTED** [review]
Looks good, added some comments. I really like the import feature that you added

File: src/Pages/Certificates/AllCerfiticates.tsx [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+        setSelectedCertificate(undefined);
+    }
+    const OnOrderByTitles = (given_order_by_text: string) => {
+        let temp_ascending = true;
+        if (order_by_text === given_order_by_text) { temp_ascending = ascending === false ? true : false; }
+        setAscending(temp_ascending)
+        setOrderByText(given_order_by_text);
+    }
+    const onSelectedEemployeeRow = (employee_id: number) => {
+        let tempCertificate
```
[the remark this answers is not in the document]
Comment (hafthorarinsson) [reply src/Pages/Certificates/AllCerfiticates.tsx:89]: As we discussed yesterday to use camelCase for variables in Typescript I'm changing the employeeId instead. Also fixed the typo in the function name

**Comment by hafthorarinsson** [comment]
@helgigretar 
Thanks for the comments. 
I've resolved all the suggestions and changed a lot of variable names from snake_case to camelCase

**Pull request merged**